modules/core/src/main/java/org/apache/ignite/internal/close/ManuallyCloseable.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.close;
+
+/**
+ * Represents something that must be eventually closed. It is different from {@link AutoCloseable} which is for being
+ * used in try-with-resources; IDEs treat any usage of an AutoCloseable outside of a try-with-resources block as a suspicious
+ * and issue a warning, so it becomes a drag to work with code that uses AutoCloseable for classes which instances
+ * are used with patterns different from the try-with-resources pattern.
+ *
+ * <p>The main reason of this interface appearance was the desire to mark 'must-be-eventually-closed' types so that
+ * we don't forget closing their instances.
```

Comment:
How does it help developers not to forget closing?

Reply:
Only as an 'annotation'. If you are unsure about whether you must close something, you might inspect its interfaces. If it is a `ManuallyCloseable`, then you must close it.

Actually, I'm not sure this is actually needed, this was suggested by @sanpwc
Sasha, could you please step in? Do we actually need it?